### PR TITLE
fix(hfb): fix weight normalization

### PR DESCRIPTION
**Pull request by arashcosmology**

### CODE REVIEW

File: ch_pipeline/hfb/analysis.py [hunk 1]
```diff
@@ -83,11 +83,11 @@ def process(self, stream):
             # Number of samples with non-zero weight in each time bin
             nsamples = np.sum(binary_weight, axis=axis_index)
 
-            # For uniform weighting, the average of the variances is
-            # sum( 1 / weight ) / nsamples,
-            # and the averaged weight is the inverse of that
+            # For uniform weighting, the variance of averaged data is
+            # sum( variance ) / (nsamples**2),
```
Comment (rikvl) [inline ch_pipeline/hfb/analysis.py:87]:
```suggestion
            # sum( variance ) / nsamples**2,
```

File: ch_pipeline/hfb/analysis.py [hunk 2]
```diff
@@ -83,11 +83,11 @@ def process(self, stream):
...
+            # and the weight is the inverse of that
             variance = tools.invert_no_zero(weight)
-            avg_weight = nsamples * tools.invert_no_zero(
+            avg_weight = (nsamples**2) * tools.invert_no_zero(
```
Comment (rikvl) [inline ch_pipeline/hfb/analysis.py:90]:
```suggestion
            avg_weight = nsamples**2 * tools.invert_no_zero(
```
I think the parentheses are not needed

File: ch_pipeline/hfb/analysis.py [hunk 3]
```diff
@@ -83,11 +83,11 @@ def process(self, stream):
             # Number of samples with non-zero weight in each time bin
             nsamples = np.sum(binary_weight, axis=axis_index)
 
-            # For uniform weighting, the average of the variances is
-            # sum( 1 / weight ) / nsamples,
-            # and the averaged weight is the inverse of that
+            # For uniform weighting, the variance of averaged data is
```
Comment (rikvl) [inline ch_pipeline/hfb/analysis.py:86]: Great! Maybe also correct the comment in line 77 from 
```
# Average the data and weights
```
to something like
```
# Average the data and compute the weight of the averaged data
```

File: ch_pipeline/hfb/analysis.py [hunk 4]
```diff
@@ -330,8 +330,10 @@ def process_finish(self):
             norm = self.stack.nsample[:].astype(np.float32)
 
             # For uniform weighting, invert the accumulated variances and
-            # multiply by number of days to finalize stack.weight.
-            self.stack.weight[:] = norm * tools.invert_no_zero(self.stack.weight[:])
+            # multiply by number of days squared to finalize stack.weight.
+            self.stack.weight[:] = (norm**2) * tools.invert_no_zero(
```
Comment (rikvl) [inline ch_pipeline/hfb/analysis.py:334]:
```suggestion
            self.stack.weight[:] = norm**2 * tools.invert_no_zero(
```

**Review by rikvl — APPROVED** [review]
Some very minor suggestions, but feel free to ignore

**Review by jrs65 — APPROVED** [review]
With Rik's changes made this one is fine.

**Pull request merged**